src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -0,0 +1,312 @@
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.6
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\View\Helper;
+
+use Cake\View\Helper;
+use Cake\View\StringTemplateTrait;
+use LogicException;
+
+/**
+ * BreadcrumbsHelper to register and display a breadcrumb trail for your views
+ *
+ * @property \Cake\View\Helper\UrlHelper $Url
+ */
+class BreadcrumbsHelper extends Helper
+{
+
+    use StringTemplateTrait;
+
+    /**
+     * Other helpers used by BreadcrumbsHelper
+     *
+     * @var array
+     */
+    public $helpers = ['Url'];
+
+    /**
+     * Default config for the helper.
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'templates' => [
+            'wrapper' => '<ul{{attrs}}>{{content}}</ul>',
+            'item' => '<li{{attrs}}><a href="{{link}}"{{innerAttrs}}>{{title}}</a></li>{{separator}}',
+            'itemWithoutLink' => '<li{{attrs}}><span{{innerAttrs}}>{{title}}</span></li>{{separator}}',
+            'separator' => '<li{{attrs}}><span{{innerAttrs}}>{{custom}}{{separator}}</span></li>'
+        ]
+    ];
+
+    /**
+     * The crumbs list.
+     *
+     * @var array
+     */
+    protected $crumbs = [];
+
+    /**
+     * Add a crumb to the trail.
+     *
+     * @param string|array $title Title of the crumb. If provided as an array, it will add each values of the array
+     * as a single crumb. This allows you to add multiple crumbs in the trail at once. Arrays are expected to be of this
+     * form:
+     * - *title* The title of the crumb
+     * - *link* The link of the crumb. If not provided, no link will be made
+     * - *options* Options of the crumb. See description of params option of this method.
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options. These options will be used as attributes HTML attribute the crumb will
+     * be rendered in (a <li> tag by default). It accepts two special keys:
+     * - *innerAttrs*: An array that allows you to define attributes for the inner element of the crumb (by default, to
+     * the link)
+     * - *templateVars*: Specific template vars in case you override the templates provided
+     * @return $this
+     */
+    public function add($title, $link = null, array $options = [])
+    {
+        if (is_array($title)) {
+            foreach ($title as $crumb) {
+                $crumb = array_merge(['title' => '', 'link' => null, 'options' => []], $crumb);
+                $this->crumbs[] = $crumb;
+            }
+
+            return $this;
+        }
+
+        $this->crumbs[] = ['title' => $title, 'link' => $link, 'options' => $options];
```

Comment:
I do wonder however if the crumbs should be a simple Breadcrumb object with basic setters and getters instead of having to deal with arrays here, but maybe thats overhead. Even the crumb list could be a BreadcumbCollection in an OOP context.